backend/src/main/java/ai/verta/modeldb/experiment/ExperimentDAORdbImpl.java

```diff
@@ -373,7 +377,8 @@ public ExperimentPaginationDTO getExperimentsInProject(
   public Experiment addExperimentTags(String experimentId, List<String> tagsList)
       throws InvalidProtocolBufferException {
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
-      ExperimentEntity experimentObj = session.load(ExperimentEntity.class, experimentId);
+      ExperimentEntity experimentObj =
+          session.load(ExperimentEntity.class, experimentId, LockMode.PESSIMISTIC_WRITE);
```

Comment:
We should not lock on read only operations or that will make the system super slow. We shouldn't even have a transaction.

Is it possible to only have pessimistic locks to write operations?

Reply:
I have added Lock on only on writable entities' operation. here it fetches but after that, it will update on https://github.com/VertaAI/modeldb/pull/1637/files/0423bed3779f3b3edc3d36d1858bd4106eb18fae#diff-5874488cfbcfa089079b8b4bc42688c13066312280f478d66f37d58b80898cc7R402 that's why I have to take a lock on it.

Reply:
sg